src/lm/kaldi-lmtable.cc

```diff
@@ -228,7 +238,13 @@ bool LmTable::ReadFstFromLmFile(std::istream &istrm,
     }
     // found, set current level
     ngram_order = atoi(inpline.substr(pos1+1, pos2-(pos1+1)).c_str());
-    cerr << "Processing " << ngram_order << "-grams" << endl;
+    std::vector<int>::iterator pos 
```

Comment:
I think it would be easier to just have an integer variable keeping track of the most recent order processed, rather than do all this messing about with vectors and iterators.  We can assume it's order 1 then order 2 then order 3, etc.